pages/instructorFileBrowser/instructorFileBrowser.ejs

```diff
@@ -12,9 +14,16 @@
   <body>
     <script>
       $(function() {
-        $('[data-toggle="popover"]').popover({
-            sanitize: false
-        })
+          $('[data-toggle="popover"]').popover({
+              sanitize: false
+          })
+
+          const modelist = ace.require('ace/ext/modelist');
```

Comment:
I'd much prefer to do this all on the server instead of on the client. You should be able to install [`ace-code`](https://www.npmjs.com/package/ace-code) and then do the following from within server-side code:

```js
const modelist = require('ace-code/src/ext/modelist');
```

Reply:
I moved the Edit disable logic to the server side (thanks for pointing to ace-code, I was struggling to get ace-builds to work server-side). For the editor itself and the choice of mode, given the JS for Ace is already used in that side, I believe it makes more sense to keep the logic that chooses the mode (rather than checks if it's editable) client side, but I'm open to discussion.